src/clj/harja/palvelin/integraatiot/velho/velho_komponentti.clj

```diff
@@ -24,10 +26,38 @@
                      ". Kesto: "
                      (float (/ (- (System/currentTimeMillis) aloitusaika-ms) 1000)) " sekuntia")))))
 
+(defn hae-varustetoteumat-velhosta [integraatioloki db asetukset]
+  (lukko/yrita-ajaa-lukon-kanssa
+    db
+    "tuo-uudet-varustetoteumat-velhosta"
+    #(suorita-ja-kirjaa-alku-loppu-ajat
+       (fn [] (varusteet/tuo-uudet-varustetoteumat-velhosta integraatioloki db asetukset))
+       "tuo-uudet-varustetoteumat-velhosta")))
+
+(defn tee-varustetoteuma-haku-tehtava-fn [{:keys [db asetukset integraatioloki]} suoritusaika]
+  (if suoritusaika
+    (do
+      (log/debug "Ajastetaan varustetoteumien tuonti Tievelhosta tehtäväksi joka päivä kello: " (-> asetukset :velho :varustetoteuma-suoritusaika))
+      (ajastettu-tehtava/ajasta-paivittain
+        suoritusaika
+        (do
+          (log/info "ajasta-paivittain :: varustetoteumien tuonti :: Alkaa " (pvm/nyt))
+          (fn [_] (hae-varustetoteumat-velhosta integraatioloki db asetukset)))))
+    (constantly nil)))
+
 (defrecord Velho [asetukset]
   component/Lifecycle
-  (start [this] this)
-  (stop [this] this)
+  (start [this]
+    (let [suoritusaika (:varuste-tuonti-suoritusaika asetukset)]
+      (log/info (str "Käynnistetään tuo-uudet-varustetoteumat-velhosta -komponentti. Suoritusaika: " suoritusaika))
+      (assoc this :varustetoteuma-tuonti-tehtava (tee-varustetoteuma-haku-tehtava-fn this suoritusaika)))
+    this)
```

Comment:
Tässä on ylimääräinen `this`, joka aiheuttaa sen, ettei `:varustetoteuma-tuonti-tehtava` tallennu komponenttiin.